src/validations/validations.ts

```diff
@@ -72,29 +98,13 @@ export function validateBoundingBoxByGrid(boundingBox: BoundingBox, referenceTil
 }
 
 /**
- * Validates that the input `geoPoint` is valid
- * @param geoPoint a point with longitude and latitude to validate
- * @param referenceTileGrid the tile grid to validate the `geoPoint` against
+ * Validates that the input `geometry` is a valid geometry
+ * @param geometry
+ * @param referenceTileGrid
  */
-export function validateGeoPoint(geoPoint: GeoPoint, referenceTileGrid: TileGrid): void {
-  if (geoPoint.lon < referenceTileGrid.boundingBox.min.lon || geoPoint.lon > referenceTileGrid.boundingBox.max.lon) {
-    throw new RangeError(`longitude ${geoPoint.lon} is out of range of tile grid's bounding box`);
-  }
-
-  if (geoPoint.lat < referenceTileGrid.boundingBox.min.lat || geoPoint.lat > referenceTileGrid.boundingBox.max.lat) {
-    throw new RangeError(`latitude ${geoPoint.lat} is out of range of tile grid's bounding box`);
-  }
-}
-
-/**
- * Validates that the input `zoom` is valid
- * @param zoom the zoom level to validate
- * @param referenceTileGrid the tile grid to validate the `zoom` against
- */
-export function validateZoomByGrid(zoom: Zoom, referenceTileGrid: TileGrid): void {
-  if (!referenceTileGrid.wellKnownScaleSet.scaleDenominators.has(zoom)) {
-    throw new Error('zoom level is not part of the given well known scale set');
-  }
+export function validateGeometryByGrid(geometry: Geometry, referenceTileGrid: TileGrid): void {
```

Comment:
I'm starting to think that the right thing to do is return a boolean value and string for reason and not throw on validation. This seems like bad practice for a library.
What do you think?

Reply:
initially validations we're supposed to be internal, as i see it, and not exposed to the lib clients. for example, calling `geoPointZoomToTile`with the wrong inputs should throw an error with a descriptive message. handling a boolean and a message by the function itself seems unnecessary and would bloat almost every exported function with this logic

Reply:
The problem is that there is no indication that the exported functions can throw errors which can lead to unexpected behavior.

Reply:
then it's just jsdoc issue
@throws should be added

Reply:
Ok, could you add this please?